src/trace/trace.c

```diff
@@ -249,7 +250,7 @@ static bool trace_filter_flood(uint32_t log_level, uint32_t entry, uint64_t mess
 	}
 
 	/* Make room for tracking new entry, by emitting the oldest one in the filter */
-	if (oldest_entry->entry_id && oldest_entry->suppress_count)
+	if (oldest_entry->entry_id && oldest_entry->trigger_count > CONFIG_TRACE_BURST_COUNT)
```

Comment:
here you definitely decide to evict the oldest entry - whatever its status was. But if its `trigger_count` wasn't greater than `CONFIG_TRACE_BURST_COUNT` you don't emit it - which makes sense, it wasn't suspended yet. But you also then don't clear the `oldest_entry` structure, so for example the `.trigger_count` field will be carried over to the new entry... And in fact you probably want to set it to 1 here, not to 0, to actually emit the configured burst count of messages, not 1 more.

Reply:
corrected